internal/shellgen/flake_package.go

```diff
@@ -0,0 +1,66 @@
+package shellgen
+
+import (
+	"path/filepath"
+	"strings"
+
+	"go.jetpack.io/devbox/internal/boxcli/featureflag"
+	"go.jetpack.io/devbox/internal/nix"
+)
+
+type flakePackage struct {
+	Name      string
+	FromStore string
+	FromPath  string
+}
+
+func newFlakePackage(pkg *nix.Package) (*flakePackage, error) {
+	// nixosCacheURL is where we fetch package binaries from
+	const nixosCacheURL = "https://cache.nixos.org"
+
+	// flakePackages only support versioned packages that have a system info
+	sysInfo := pkg.SystemInfo()
+	if sysInfo == nil {
+		return nil, nil
+	}
+
+	attributePath, err := pkg.PackageAttributePath()
+	if err != nil {
+		return nil, err
+	}
+
+	storeDir := strings.Join([]string{sysInfo.FromHash, sysInfo.StoreName, sysInfo.StoreVersion}, "-")
+	return &flakePackage{
+		Name:      attributePath,
+		FromStore: nixosCacheURL,
+		FromPath:  filepath.Join("/nix/store", storeDir),
+		// TODO add ToPath:
```

Comment:
Have you looked at the [`builtins.storePath` function](https://nixos.org/manual/nix/stable/language/builtins.html#builtins-storePath)? If we're running in impure mode, then it might make things a little easier. It doesn't require a content-addressed hash and it will automatically use the Nix-configured binary caches (aka substituters). That'll make this work automatically with nixed in Devbox Cloud.